### **Paso 5.2.8 - Ingesta del directorio "qualifying" para el directorio "2021-03-21"**

Nos permite crear e indicar parámetros en tiempo de ejecución

<center><img src="https://i.postimg.cc/4NhmHcbQ/db152.png"></center>

In [ ]:
dbutils.widgets.text("p_data_source", "")
v_data_source = dbutils.widgets.get("p_data_source")

In [ ]:
v_data_source

Out[3]: 'Eargast'

In [ ]:
dbutils.widgets.text("p_file_date", "2021-03-21")
v_file_date = dbutils.widgets.get("p_file_date")

In [ ]:
v_file_date

Out[5]: '2021-03-21'

In [ ]:
%run "../includes/configuration"

In [ ]:
%run "../includes/common_functions"

#### Paso 1 - Leer el directorio **qualifying** el cual contiene multiples archivos Multi Line JSON

In [ ]:
from pyspark.sql.types import StructType, StructField, IntegerType, StringType

In [ ]:
qualifying_schema = StructType(fields=[StructField("qualifyId", IntegerType(), False),
                                      StructField("raceId", IntegerType(), True),
                                      StructField("driverId", IntegerType(), True),
                                      StructField("constructorId", IntegerType(), True),
                                      StructField("number", IntegerType(), True),
                                      StructField("position", IntegerType(), True),
                                      StructField("q1", StringType(), True),
                                      StructField("q2", StringType(), True),
                                      StructField("q3", StringType(), True),
                                     ])

In [ ]:
# El parámetro "raw_folder_path" se encuentra en el notebook "configuration"
# El parámetro "v_file_date" se encuentra en el notebook e indicamos su valor en tiempo de ejecución
qualifying_df = spark.read \
.schema(qualifying_schema) \
.option("multiLine", True) \
.json(f"{raw_folder_path}/{v_file_date}/qualifying")
# Esto seria equivalente a la ruta: /mnt/formula1dl/raw/2021-03-21/qualifying

In [ ]:
qualifying_df.show(truncate=False)

+---------+------+--------+-------------+------+--------+--------+--------+--------+
|qualifyId|raceId|driverId|constructorId|number|position|q1      |q2      |q3      |
+---------+------+--------+-------------+------+--------+--------+--------+--------+
|1        |18    |1       |1            |22    |1       |1:26.572|1:25.187|1:26.714|
|2        |18    |9       |2            |4     |2       |1:26.103|1:25.315|1:26.869|
|3        |18    |5       |1            |23    |3       |1:25.664|1:25.452|1:27.079|
|4        |18    |13      |6            |2     |4       |1:25.994|1:25.691|1:27.178|
|5        |18    |2       |2            |3     |5       |1:25.960|1:25.518|1:27.236|
|6        |18    |15      |7            |11    |6       |1:26.427|1:26.101|1:28.527|
|7        |18    |3       |3            |7     |7       |1:26.295|1:26.059|1:28.687|
|8        |18    |14      |9            |9     |8       |1:26.381|1:26.063|1:29.041|
|9        |18    |10      |7            |12    |9       |1:26.919

#### Paso 2 - Renombrar columnas y añadir nuevas columnas
1. Renombrar qualifyingId, driverId, constructorId y raceId
2. Añadir ingestion_date con current timestamp

In [ ]:
from pyspark.sql.functions import lit

In [ ]:
# La función "add_ingestion_date()" se encuentra en el notebook "common_functions"
qualifying_with_ingestion_date_df = add_ingestion_date(qualifying_df)

In [ ]:
qualifying_with_ingestion_date_df.show(truncate=False)

+---------+------+--------+-------------+------+--------+--------+--------+--------+-----------------------+
|qualifyId|raceId|driverId|constructorId|number|position|q1      |q2      |q3      |ingestion_date         |
+---------+------+--------+-------------+------+--------+--------+--------+--------+-----------------------+
|1        |18    |1       |1            |22    |1       |1:26.572|1:25.187|1:26.714|2023-06-15 17:43:10.249|
|2        |18    |9       |2            |4     |2       |1:26.103|1:25.315|1:26.869|2023-06-15 17:43:10.249|
|3        |18    |5       |1            |23    |3       |1:25.664|1:25.452|1:27.079|2023-06-15 17:43:10.249|
|4        |18    |13      |6            |2     |4       |1:25.994|1:25.691|1:27.178|2023-06-15 17:43:10.249|
|5        |18    |2       |2            |3     |5       |1:25.960|1:25.518|1:27.236|2023-06-15 17:43:10.249|
|6        |18    |15      |7            |11    |6       |1:26.427|1:26.101|1:28.527|2023-06-15 17:43:10.249|
|7        |18    |3

In [ ]:
final_df = qualifying_with_ingestion_date_df.withColumnRenamed("qualifyId", "qualify_id") \
                                            .withColumnRenamed("driverId", "driver_id") \
                                            .withColumnRenamed("raceId", "race_id") \
                                            .withColumnRenamed("constructorId", "constructor_id") \
                                            .withColumn("data_source", lit(v_data_source)) \
                                            .withColumn("file_date", lit(v_file_date))

In [ ]:
final_df.show(truncate=False)

+----------+-------+---------+--------------+------+--------+--------+--------+--------+-----------------------+-----------+----------+
|qualify_id|race_id|driver_id|constructor_id|number|position|q1      |q2      |q3      |ingestion_date         |data_source|file_date |
+----------+-------+---------+--------------+------+--------+--------+--------+--------+-----------------------+-----------+----------+
|1         |18     |1        |1             |22    |1       |1:26.572|1:25.187|1:26.714|2023-06-15 17:43:10.968|Eargast    |2021-03-21|
|2         |18     |9        |2             |4     |2       |1:26.103|1:25.315|1:26.869|2023-06-15 17:43:10.968|Eargast    |2021-03-21|
|3         |18     |5        |1             |23    |3       |1:25.664|1:25.452|1:27.079|2023-06-15 17:43:10.968|Eargast    |2021-03-21|
|4         |18     |13       |6             |2     |4       |1:25.994|1:25.691|1:27.178|2023-06-15 17:43:10.968|Eargast    |2021-03-21|
|5         |18     |2        |2             |3  

In [ ]:
final_df.printSchema()

root
 |-- qualify_id: integer (nullable = true)
 |-- race_id: integer (nullable = true)
 |-- driver_id: integer (nullable = true)
 |-- constructor_id: integer (nullable = true)
 |-- number: integer (nullable = true)
 |-- position: integer (nullable = true)
 |-- q1: string (nullable = true)
 |-- q2: string (nullable = true)
 |-- q3: string (nullable = true)
 |-- ingestion_date: timestamp (nullable = false)
 |-- data_source: string (nullable = false)
 |-- file_date: string (nullable = false)



#### Paso 3 - Escribir datos en el datalake como parquet y crear la tabla **qualifying** en la base de datos **f1_processed**

In [ ]:
final_df.select('race_id').distinct().collect()

Out[18]: [Row(race_id=243),
 Row(race_id=858),
 Row(race_id=31),
 Row(race_id=85),
 Row(race_id=251),
 Row(race_id=65),
 Row(race_id=53),
 Row(race_id=255),
 Row(race_id=78),
 Row(race_id=857),
 Row(race_id=108),
 Row(race_id=34),
 Row(race_id=193),
 Row(race_id=211),
 Row(race_id=115),
 Row(race_id=101),
 Row(race_id=81),
 Row(race_id=847),
 Row(race_id=28),
 Row(race_id=210),
 Row(race_id=842),
 Row(race_id=76),
 Row(race_id=26),
 Row(race_id=27),
 Row(race_id=44),
 Row(race_id=192),
 Row(race_id=271),
 Row(race_id=844),
 Row(race_id=253),
 Row(race_id=103),
 Row(race_id=12),
 Row(race_id=350),
 Row(race_id=860),
 Row(race_id=91),
 Row(race_id=22),
 Row(race_id=209),
 Row(race_id=230),
 Row(race_id=122),
 Row(race_id=225),
 Row(race_id=93),
 Row(race_id=246),
 Row(race_id=346),
 Row(race_id=855),
 Row(race_id=861),
 Row(race_id=224),
 Row(race_id=111),
 Row(race_id=47),
 Row(race_id=140),
 Row(race_id=177),
 Row(race_id=353),
 Row(race_id=355),
 Row(race_id=259),
 Row(race_id=1),
 Ro

In [ ]:
for race_id_list in final_df.select('race_id').distinct().collect():
  print(race_id_list)

Row(race_id=243)
Row(race_id=858)
Row(race_id=31)
Row(race_id=85)
Row(race_id=251)
Row(race_id=65)
Row(race_id=53)
Row(race_id=255)
Row(race_id=78)
Row(race_id=857)
Row(race_id=108)
Row(race_id=34)
Row(race_id=193)
Row(race_id=211)
Row(race_id=115)
Row(race_id=101)
Row(race_id=81)
Row(race_id=847)
Row(race_id=28)
Row(race_id=210)
Row(race_id=842)
Row(race_id=76)
Row(race_id=26)
Row(race_id=27)
Row(race_id=44)
Row(race_id=192)
Row(race_id=271)
Row(race_id=844)
Row(race_id=253)
Row(race_id=103)
Row(race_id=12)
Row(race_id=350)
Row(race_id=860)
Row(race_id=91)
Row(race_id=22)
Row(race_id=209)
Row(race_id=230)
Row(race_id=122)
Row(race_id=225)
Row(race_id=93)
Row(race_id=246)
Row(race_id=346)
Row(race_id=855)
Row(race_id=861)
Row(race_id=224)
Row(race_id=111)
Row(race_id=47)
Row(race_id=140)
Row(race_id=177)
Row(race_id=353)
Row(race_id=355)
Row(race_id=259)
Row(race_id=1)
Row(race_id=52)
Row(race_id=212)
Row(race_id=13)
Row(race_id=348)
Row(race_id=86)
Row(race_id=6)
Row(race_id=16)
Row(r

In [ ]:
for race_id_list in final_df.select('race_id').distinct().collect():
  print(race_id_list.race_id)

243
858
31
85
251
65
53
255
78
857
108
34
193
211
115
101
81
847
28
210
842
76
26
27
44
192
271
844
253
103
12
350
860
91
22
209
230
122
225
93
246
346
855
861
224
111
47
140
177
353
355
259
1
52
212
13
348
86
6
16
852
3
20
40
340
250
94
57
339
54
120
96
846
48
266
191
268
5
258
19
92
257
64
227
117
41
347
43
15
845
262
269
112
849
843
854
207
37
61
88
263
107
9
17
72
175
850
35
229
173
114
55
59
4
241
8
100
161
23
39
49
176
7
244
84
856
247
87
343
51
69
97
264
63
77
10
102
848
50
267
45
216
38
261
82
249
80
240
25
245
73
113
24
160
70
62
125
121
156
351
95
29
226
21
214
338
352
256
98
345
32
60
90
75
203
265
341
859
200
272
56
109
254
213
228
354
105
58
33
11
83
110
68
106
71
116
14
344
349
342
123
158
199
242
270
248
42
337
119
79
201
2
118
30
99
66
46
67
215
252
851
18
74
104
841
36
208
89
897
1025
879
883
898
970
853
918
1005
1016
961
974
876
976
950
874
939
1031
1030
1034
988
914
1019
908
916
973
984
1046
926
875
995
882
906
955
912
896
1008
992
1047
887
972
998
867
881
980
1021
8

In [ ]:
n = 0
for race_id_list in final_df.select('race_id').distinct().collect():
  n = n + 1
print(n)

404


1. Si no hubiesemos utilizado la sentencia **IF** y la tabla **f1_processed.qualifying** no hubiese existido, nos hubiese devuelto un **ERROR**. Es por eso que utilizamos el **IF**
2. **spark.conf.set("spark.sql.sources.partitionOverwriteMode","dynamic")** nos permite utilizar la función **insertInto** en modo **overwrite** y asi SOLO SOBRESCRIBIR las particiones que sean coincidentes. Tengo un ejemplo en mi material de **Apache Spark con Python**
3. Por lo tanto, si la tabla ya existe, solo sobrescribirá las particiones coincidentes
4. Y si no existe la tabla, la creará
5. Como estamos ingestando el directorio de archivos JSON **qualifying** del directorio **2021-03-21**, existen 404 particiones
6. Con este método, podemos EJECUTAR EL NOTEBOOK CUANTAS VECES QUERAMOS Y NO SE DUPLICARAN LOS DATOS

In [ ]:
spark.conf.set("spark.sql.sources.partitionOverwriteMode","dynamic")

In [ ]:
final_df = final_df.select("qualify_id","driver_id","constructor_id","number","position","q1","q2","q3","ingestion_date","data_source","file_date","race_id")

In [ ]:
if (spark._jsparkSession.catalog().tableExists("f1_processed.qualifying")):
  final_df.write.mode('overwrite').insertInto("f1_processed.qualifying")
else:
  final_df.write.partitionBy('race_id').format('parquet').saveAsTable("f1_processed.qualifying")

In [ ]:
spark.read.parquet("/mnt/formula1dl/processed/qualifying").show(truncate=False)

+----------+---------+--------------+------+--------+--------+--------+--------+-----------------------+-----------+----------+-------+
|qualify_id|driver_id|constructor_id|number|position|q1      |q2      |q3      |ingestion_date         |data_source|file_date |race_id|
+----------+---------+--------------+------+--------+--------+--------+--------+-----------------------+-----------+----------+-------+
|3374      |17       |9             |6     |1       |1:51.886|1:48.210|1:49.327|2023-06-15 17:43:16.777|Eargast    |2021-03-21|339    |
|3375      |3        |131           |4     |2       |1:52.560|1:47.417|1:50.673|2023-06-15 17:43:16.777|Eargast    |2021-03-21|339    |
|3376      |20       |9             |5     |3       |1:47.632|1:46.828|1:50.789|2023-06-15 17:43:16.777|Eargast    |2021-03-21|339    |
|3377      |16       |10            |14    |4       |1:49.479|1:47.085|1:50.914|2023-06-15 17:43:16.777|Eargast    |2021-03-21|339    |
|3378      |807      |3             |10    |5   

In [ ]:
%sql
SELECT * FROM f1_processed.qualifying;

qualify_id driver_id constructor_id number position q1 q2 q3 ingestion_date data_source file_date race_id 3374 17 9 6 1 1:51.886 1:48.210 1:49.327 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3375 3 131 4 2 1:52.560 1:47.417 1:50.673 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3376 20 9 5 3 1:47.632 1:46.828 1:50.789 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3377 16 10 14 4 1:49.479 1:47.085 1:50.914 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3378 807 3 10 5 1:49.664 1:47.346 1:51.001 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3379 9 4 11 6 1:46.283 1:46.951 1:51.051 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3380 22 3 9 7 1:50.301 1:48.371 1:51.511 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3381 30 131 3 8 1:52.239 1:48.400 1:51.717 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3382 155 15 23 9 1:48.467 1:47.792 1:51.767 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3383 24 10 15 10 1:49.922 1:48.238 1:52.254 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3384 808 4 12 11 1:47.952 1:48.760 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3385 37 15 22 12 1:47.153 1:48.771 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3386 67 5 16 13 1:48.945 1:49.207 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3387 153 5 17 14 1:48.655 1:49.464 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3388 5 205 19 15 1:52.875 1:52.270 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3389 10 166 24 16 1:52.398 1:52.520 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3390 18 1 1 17 1:52.211 \N \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3391 15 205 18 18 1:52.884 \N \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3392 4 6 8 19 1:53.044 \N \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3393 1 1 2 20 1:53.050 \N \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3394 13 6 7 21 1:53.283 \N \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3395 812 164 20 22 1:56.299 \N \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3396 811 164 21 23 1:57.269 \N \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 3397 810 166 25 24 1:59.977 \N \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 339 4562 20 9 1 1 1:37.468 1:35.934 1:34.870 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4563 1 1 3 2 1:36.861 1:35.852 1:34.974 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4564 17 9 2 3 1:37.924 1:36.080 1:35.179 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4565 18 1 4 4 1:37.033 1:35.569 1:35.200 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4566 4 6 5 5 1:36.897 1:36.320 1:35.802 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4567 2 4 9 6 1:37.224 1:36.811 1:36.124 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4568 13 6 6 7 1:36.744 1:36.557 1:36.251 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4569 808 4 10 8 1:37.210 1:36.642 1:36.324 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4570 3 131 8 9 1:37.316 1:36.388 1:36.809 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4571 155 15 16 10 1:36.994 1:36.691 1:36.820 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4572 30 131 7 11 1:36.904 1:37.035 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4573 67 5 18 12 1:37.693 1:37.160 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4574 153 5 19 13 1:37.677 1:37.347 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4575 814 10 15 14 1:38.045 1:37.370 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4576 22 3 11 15 1:38.163 1:37.496 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4577 815 15 17 16 1:37.759 1:37.528 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4578 16 10 14 17 1:37.693 1:37.593 \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4579 813 3 12 18 1:38.276 \N \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4580 5 205 20 19 1:38.645 \N \N 2023-06-15T17:43:16.777+0000 Eargast 2021-03-21 842 4581 15 20

In [ ]:
%sql
-- Vemos que tenemos la data hasta la race_id = 1047
SELECT race_id, COUNT(1)
FROM f1_processed.qualifying
GROUP BY race_id
ORDER BY race_id DESC

race_id count(1) 1047 20 1046 20 1045 20 1044 20 1043 20 1042 20 1041 20 1040 20 1039 20 1038 20 1037 20 1036 20 1035 20 1034 20 1033 20 1032 20 1031 20 1030 20 1029 20 1028 20 1027 20 1026 20 1025 20 1024 20 1023 20 1022 20 1021 20 1020 20 1019 20 1018 20 1017 20 1016 20 1015 20 1014 20 1013 20 1012 18 1011 20 1010 20 1009 20 1008 20 1007 20 1006 20 1005 20 1004 20 1003 20 1002 20 1001 20 1000 20 999 20 998 20 997 20 996 20 995 20 994 20 993 20 992 20 991 20 990 20 989 20 988 20 987 20 986 19 985 20 984 20 983 20 982 20 981 20 980 20 979 20 978 20 977 20 976 19 975 20 974 20 973 20 972 20 971 20 970 20 969 20 968 22 967 22 966 21 965 22 964 22 963 22 962 22 961 21 960 21 959 22 958 22 957 21 956 22 955 22 954 21 953 22 952 22 951 22 950 22 949 22 948 22 945 20 944 20 943 20 942 20 941 19 940 20 939 20 938 20 937 20 936 20 934 20 933 20 932 19 931 20 930 20 929 19 928 20 927 19 926 18 918 18 917 18 916 18 915 21 914 22 913 22 912 22 911 22 910 22 909 22 908 22 907 22 906 22 905 22 904 22 903 22 902 22 901 22 900 22 899 22 898 22 897 22 896 22 895 22 894 22 893 22 892 22 891 22 890 22 888 22 887 22 886 22 885 22 884 22 883 22 882 22 881 22 880 22 879 24 878 24 877 24 876 24 875 23 874 24 873 24 872 23 871 24 870 24 869 24 868 24 867 23 866 24 865 23 864 24 863 24 862 24 861 24 860 24 859 24 858 23 857 24 856 24 855 24 854 24 853 24 852 23 851 24 850 24 849 24 848 24 847 24 846 24 845 23 844 23 843 24 842 24 841 24 355 24 354 24 353 24 352 24 351 24 350 24 349 24 348 24 347 24 346 24 345 24 344 24 343 24 342 24 341 24 340 24 339 24 338 24 337 24 272 26 271 26 270 26 269 26 268 26 267 26 266 26 265 26 264 26 263 26 262 26 261 26 259 28 258 26 257 26 256 24 255 24 254 24 253 24 252 24 251 24 250 24 249 24 248 24 247 24 246 24 245 24 244 26 243 26 242 26 241 26 240 26 230 20 229 22 228 20 227 20 226 22 225 22 224 20 216 22 215 22 214 22 213 22 212 22 211 22 210 22 209 22 208 22 207 22 203 22 201 21 200 22 199 22 193 22 192 22 191 22 177 22 176 21 175 22 173 22 161 22 160 22 158 22 156 22 140 20 125 22 123 20 122 20 121 20 120 20 119 20 118 20 117 20 116 20 115 20 114 20 113 20 112 20 111 20 110 20 109 20 108 20 107 20 106 20 105 20 104 20 103 20 102 20 101 20 100 20 99 20 98 20 97 20 96 20 95 20 94 20 93 20 92 20 91 20 90 20 89 20 88 20 87 20 86 20 85 20 84 20 83 20 82 20 81 20 80 20 79 20 78 20 77 20 76 18 75 18 74 20 73 20 72 20 71 20 70 22 69 22 68 22 67 22 66 22 65 22 64 22 63 22 62 22 61 22 60 22 59 22 58 22 57 22 56 22 55 22 54 22 53 22 52 22 51 22 50 22 49 22 48 22 47 22 46 22 45 22 44 22 43 22 42 22 41 22 40 22 39 22 38 22 37 22 36 22 35 20 34 20 33 20 32 20 31 20 30 20 29 20 28 20 27 20 26 20 25 20 24 20 23 20 22 20 21 22 20 22 19 22 18 22 17 20 16 20 15 20 14 20 13 20 12 20 11 20 10 20 9 20 8 20 7 20 6 20 5 20 4 20 3 20 2 20 1 20

In [ ]:
%sql
DESCRIBE FORMATTED f1_processed.qualifying;

col_name data_type comment qualify_id int null driver_id int null constructor_id int null number int null position int null q1 string null q2 string null q3 string null ingestion_date timestamp null data_source string null file_date string null race_id int null # Partition Information # col_name data_type comment race_id int null # Detailed Table Information Catalog spark_catalog Database f1_processed Table qualifying Owner root Created Time Thu Jun 15 17:44:31 UTC 2023 Last Access UNKNOWN Created By Spark 3.3.2 Type MANAGED Provider parquet Location dbfs:/mnt/formula1dl/processed/qualifying Serde Library org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe InputFormat org.apache.hadoop.mapred.SequenceFileInputFormat OutputFormat org.apache.hadoop.hive.ql.io.HiveSequenceFileOutputFormat Partition Provider Catalog

<center><img src="https://i.postimg.cc/mr6g0vT8/db120.png"></center>

In [ ]:
dbutils.notebook.exit("Success")

Success